File: src/frontend/types/patient.ts
export interface PatientResponse {
  id: string;
  tenantId: string;
  firstName: string;
  lastName: string;
  phoneNumber: string;
  primaryBranchId: string | null;
  primaryBranchName: string | null;
  createdAt: string;
}

export interface CreatePatientRequest {
  tenantId: string;
  firstName: string;
  lastName: string;
  phoneNumber: string;
  primaryBranchId?: string;
}

File: src/backend/Entities/Patient.cs
namespace ClinicPos.Api.Entities;

public class Patient : ITenantScoped
{
    public Guid Id { get; set; }
    public Guid TenantId { get; set; }
    public string FirstName { get; set; } = string.Empty;
    public string LastName { get; set; } = string.Empty;
    public string PhoneNumber { get; set; } = string.Empty;
    public Guid? PrimaryBranchId { get; set; }
    public DateTime CreatedAt { get; set; }
    public DateTime UpdatedAt { get; set; }
    public bool IsDeleted { get; set; }

    public Tenant Tenant { get; set; } = null!;
    public Branch? PrimaryBranch { get; set; }
}
